comms/src/builder/builder.rs

```diff
@@ -434,58 +433,59 @@ impl CommsServicesHandle {
     }
 }
 
-#[cfg(test)]
-mod test {
-    use super::*;
-
-    mod handlers {
-        use super::*;
-        use crate::{dispatcher::HandlerError, message::DomainMessageContext};
-        use serde::Deserialize;
-
-        #[derive(Serialize, Deserialize)]
-        pub struct TestMessage {
-            name: String,
-            age: u8,
-        }
-
-        pub fn hello(context: DomainMessageContext) -> Result<(), HandlerError> {
-            let msg: TestMessage = context.message.to_message().map_err(HandlerError::failed())?;
-            debug!("Hello: {:?}, you are {} years old", msg.name, msg.age);
-            Ok(())
-        }
-
-        pub fn catch_all(_msg: DomainMessageContext) -> Result<(), HandlerError> {
-            Ok(())
-        }
-    }
-
-    #[test]
-    fn new_no_control_service() {
-        let comms_services = CommsBuilder::new(|| {
-            DomainMessageDispatcher::default()
-                .route("hello".to_owned(), handlers::hello)
-                .catch_all(handlers::catch_all)
-        })
-        .with_node_identity(NodeIdentity::random_for_test(None))
-        .build()
-        .unwrap();
-
-        assert!(comms_services.control_service.is_none());
-    }
-
-    #[test]
-    fn new_with_control_service() {
-        let comms_services = CommsBuilder::new(|| {
-            DomainMessageDispatcher::default()
-                .route("hello".to_owned(), handlers::hello)
-                .catch_all(handlers::catch_all)
-        })
-        .with_node_identity(NodeIdentity::random_for_test(None))
-        .configure_control_service(|| ControlServiceConfig::default())
-        .build()
-        .unwrap();
-
-        assert!(comms_services.control_service.is_some());
-    }
-}
+// TODO: The handler functions need to be changed to handler services for these tests to work
```

Comment:
No problem, I'll be working on wiring the builder up with the message broker and fixing the tests 👍

Reply:
Perfect, thanx